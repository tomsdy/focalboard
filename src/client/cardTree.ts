import { Card } from "./card"
import { OctoClient } from "./octoClient"
import { IBlock } from "./octoTypes"

class CardTree {
	card: Card
	comments: IBlock[]
	contents: IBlock[]
	isSynched: boolean

	constructor(
		private octo: OctoClient,
		private cardId: string) {
	}

	async sync() {
		const blocks = await this.octo.getSubtree(this.cardId)
		this.rebuild(blocks)
	}

	private rebuild(blocks: IBlock[]) {
		this.card = new Card(blocks.find(o => o.id === this.cardId))

		this.comments = blocks
			.filter(block => block.type === "comment")
			.sort((a, b) => a.createAt - b.createAt)

		this.contents = blocks
			.filter(block => block.type === "text" || block.type === "image")
			.sort((a, b) => a.order - b.order)

		this.isSynched = true
	}
}

export { CardTree }
