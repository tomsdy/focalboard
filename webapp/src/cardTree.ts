// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {Block} from './blocks/block';
import {Card} from './blocks/card';
import octoClient from './octoClient';
import {IBlock, IOrderedBlock} from './octoTypes';
import {OctoUtils} from './octoUtils';

class CardTree {
    card: Card
    comments: IBlock[]
    contents: IOrderedBlock[]
    isSynched: boolean

    constructor(private cardId: string) {
    }

    async sync() {
	    const blocks = await octoClient.getSubtree(this.cardId)
        this.rebuild(OctoUtils.hydrateBlocks(blocks))
    }

    private rebuild(blocks: Block[]) {
	    this.card = blocks.find((o) => o.id === this.cardId) as Card

        this.comments = blocks.
            filter((block) => block.type === 'comment').
            sort((a, b) => a.createAt - b.createAt)

	    const contentBlocks = blocks.filter((block) => block.type === 'text' || block.type === 'image') as IOrderedBlock[]
        this.contents = contentBlocks.sort((a, b) => a.order - b.order)

        this.isSynched = true
    }
}

export {CardTree}
