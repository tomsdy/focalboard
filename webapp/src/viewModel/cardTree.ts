// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {ContentBlockTypes, contentBlockTypes, IBlock} from '../blocks/block'
import {Card, MutableCard} from '../blocks/card'
import {CommentBlock} from '../blocks/commentBlock'
import {IContentBlock} from '../blocks/contentBlock'
import octoClient from '../octoClient'
import {OctoUtils} from '../octoUtils'

interface CardTree {
    readonly card: Card
    readonly comments: readonly CommentBlock[]
    readonly contents: readonly IContentBlock[]
    readonly allBlocks: readonly IBlock[]
}

class MutableCardTree implements CardTree {
    card: MutableCard
    comments: CommentBlock[] = []
    contents: IContentBlock[] = []

    get allBlocks(): IBlock[] {
        return [this.card, ...this.comments, ...this.contents]
    }

    constructor(card: MutableCard) {
        this.card = card
    }

    // Factory methods

    static async sync(boardId: string): Promise<CardTree | undefined> {
        const rawBlocks = await octoClient.getSubtree(boardId)
        return this.buildTree(boardId, rawBlocks)
    }

    static incrementalUpdate(cardTree: CardTree, updatedBlocks: IBlock[]): CardTree | undefined {
        const relevantBlocks = updatedBlocks.filter((block) => block.deleteAt !== 0 || block.id === cardTree.card.id || block.parentId === cardTree.card.id)
        if (relevantBlocks.length < 1) {
            // No change
            return cardTree
        }
        const rawBlocks = OctoUtils.mergeBlocks(cardTree.allBlocks, relevantBlocks)
        return this.buildTree(cardTree.card.id, rawBlocks)
    }

    private static buildTree(cardId: string, sourceBlocks: readonly IBlock[]): MutableCardTree | undefined {
        const blocks = OctoUtils.hydrateBlocks(sourceBlocks)

        const card = blocks.find((o) => o.type === 'card' && o.id === cardId) as MutableCard
        if (!card) {
            return undefined
        }
        const cardTree = new MutableCardTree(card)
        cardTree.comments = blocks.
            filter((block) => block.type === 'comment').
            sort((a, b) => a.createAt - b.createAt) as CommentBlock[]

        const contentBlocks = blocks.filter((block) => contentBlockTypes.includes(block.type as ContentBlockTypes)) as IContentBlock[]
        cardTree.contents = OctoUtils.getBlockOrder(card.contentOrder, contentBlocks)

        return cardTree
    }

    // private mutableCopy(): MutableCardTree {
    //     return MutableCardTree.buildTree(this.card.id, this.allBlocks)!
    // }
}

export {MutableCardTree, CardTree}
