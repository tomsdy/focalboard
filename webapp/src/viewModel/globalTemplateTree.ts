// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {IBlock} from '../blocks/block'
import {Board} from '../blocks/board'
import octoClient, {OctoClient} from '../octoClient'
import {OctoUtils} from '../octoUtils'

interface GlobalTemplateTree {
    readonly boardTemplates: readonly Board[]
    readonly allBlocks: readonly IBlock[]
}

class MutableGlobalTemplateTree implements GlobalTemplateTree {
    boardTemplates: Board[] = []
    get allBlocks(): IBlock[] {
        return [...this.boardTemplates]
    }

    // Factory methods

    static async sync(): Promise<GlobalTemplateTree> {
        const rootClient = new OctoClient(octoClient.serverUrl, '0')
        const rawBlocks = await rootClient.getBlocksWithType('board')

        return this.buildTree(rawBlocks)
    }

    static incrementalUpdate(originalTree: GlobalTemplateTree, updatedBlocks: IBlock[]): GlobalTemplateTree {
        const relevantBlocks = updatedBlocks.filter((block) => block.deleteAt !== 0 || block.type === 'board' || block.type === 'view')
        if (relevantBlocks.length < 1) {
            // No change
            return originalTree
        }
        const rawBlocks = OctoUtils.mergeBlocks(originalTree.allBlocks, relevantBlocks)
        return this.buildTree(rawBlocks)
    }

    private static buildTree(sourceBlocks: readonly IBlock[]): MutableGlobalTemplateTree {
        const blocks = OctoUtils.hydrateBlocks(sourceBlocks)

        const workspaceTree = new MutableGlobalTemplateTree()
        const allBoards = blocks.filter((block) => block.type === 'board') as Board[]
        workspaceTree.boardTemplates = allBoards.filter((block) => block.isTemplate).
            sort((a, b) => a.title.localeCompare(b.title)) as Board[]

        return workspaceTree
    }

    // private mutableCopy(): MutableWorkspaceTree {
    //     return MutableWorkspaceTree.buildTree(this.allBlocks)!
    // }
}

export {MutableGlobalTemplateTree, GlobalTemplateTree}
