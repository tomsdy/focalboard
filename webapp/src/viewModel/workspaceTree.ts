// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {Board} from '../blocks/board'
import octoClient from '../octoClient'
import {IBlock} from '../blocks/block'
import {OctoUtils} from '../octoUtils'
import {BoardView} from '../blocks/boardView'

interface WorkspaceTree {
    readonly boards: readonly Board[]
    readonly views: readonly BoardView[]

    mutableCopy(): MutableWorkspaceTree
}

class MutableWorkspaceTree {
    boards: Board[] = []
    views: BoardView[] = []

    private rawBlocks: IBlock[] = []

    async sync() {
        const rawBoards = await octoClient.getBlocksWithType('board')
        const rawViews = await octoClient.getBlocksWithType('view')
        this.rawBlocks = [...rawBoards, ...rawViews]
        this.rebuild(OctoUtils.hydrateBlocks(this.rawBlocks))
    }

    incrementalUpdate(updatedBlocks: IBlock[]) {
        const relevantBlocks = updatedBlocks.filter((block) => block.type === 'board' || block.type === 'view')
        if (relevantBlocks.length < 1) {
            return
        }
        this.rawBlocks = OctoUtils.mergeBlocks(this.rawBlocks, updatedBlocks)
        this.rebuild(OctoUtils.hydrateBlocks(this.rawBlocks))
    }

    private rebuild(blocks: IBlock[]) {
        this.boards = blocks.filter((block) => block.type === 'board') as Board[]
        this.views = blocks.filter((block) => block.type === 'view') as BoardView[]
    }

    mutableCopy(): MutableWorkspaceTree {
        const workspaceTree = new MutableWorkspaceTree()
        workspaceTree.incrementalUpdate(this.rawBlocks)
        return workspaceTree
    }
}

export {MutableWorkspaceTree, WorkspaceTree}
