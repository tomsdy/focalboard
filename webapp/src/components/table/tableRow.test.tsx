// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react'
import {render} from '@testing-library/react'
import '@testing-library/jest-dom'
import {IntlProvider} from 'react-intl'

import 'isomorphic-fetch'

import {DndProvider} from 'react-dnd'
import {HTML5Backend} from 'react-dnd-html5-backend'

import {TestBlockFactory} from '../../test/testBlockFactory'
import {FetchMock} from '../../test/fetchMock'
import {MutableBoardTree} from '../../viewModel/boardTree'

import {MutableCardTree} from '../../viewModel/cardTree'

import TableRow from './tableRow'

global.fetch = FetchMock.fn

beforeEach(() => {
    FetchMock.fn.mockReset()
})

const wrapProviders = (children: any) => {
    return (
        <DndProvider backend={HTML5Backend}>
            <IntlProvider locale='en'>{children}</IntlProvider>
        </DndProvider>
    )
}

describe('components/table/TableRow', () => {
    const board = TestBlockFactory.createBoard()
    const view = TestBlockFactory.createBoardView(board)

    const view2 = TestBlockFactory.createBoardView(board)
    view2.sortOptions = []

    const card = TestBlockFactory.createCard(board)
    const cardTemplate = TestBlockFactory.createCard(board)
    cardTemplate.isTemplate = true

    test('should match snapshot', async () => {
        // Sync
        FetchMock.fn.mockReturnValueOnce(FetchMock.jsonResponse(JSON.stringify([board, view, view2, card, cardTemplate])))
        FetchMock.fn.mockReturnValue(FetchMock.jsonResponse(JSON.stringify([{username: 'username_1'}, {username: 'username_2'}])))

        const boardTree = await MutableBoardTree.sync(board.id, view.id)
        expect(boardTree).toBeDefined()
        expect(FetchMock.fn).toBeCalledTimes(2)

        const cardTree = new MutableCardTree(card)

        const component = wrapProviders(
            <TableRow
                boardTree={boardTree!}
                cardTree={cardTree}
                card={card}
                isSelected={false}
                focusOnMount={false}
                onSaveWithEnter={jest.fn()}
                showCard={jest.fn()}

                readonly={false}
                offset={0}
                resizingColumn={''}
                columnRefs={new Map()}
                onDrop={jest.fn()}
            />,
        )
        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })

    test('should match snapshot, read-only', async () => {
        // Sync
        FetchMock.fn.mockReturnValueOnce(FetchMock.jsonResponse(JSON.stringify([board, view, view2, card, cardTemplate])))
        FetchMock.fn.mockReturnValue(FetchMock.jsonResponse(JSON.stringify([{username: 'username_1'}, {username: 'username_2'}])))

        const boardTree = await MutableBoardTree.sync(board.id, view.id)
        expect(boardTree).toBeDefined()
        expect(FetchMock.fn).toBeCalledTimes(2)

        const cardTree = new MutableCardTree(card)

        const component = wrapProviders(
            <TableRow
                boardTree={boardTree!}
                cardTree={cardTree}
                card={card}
                isSelected={false}
                focusOnMount={false}
                onSaveWithEnter={jest.fn()}
                showCard={jest.fn()}

                readonly={true}
                offset={0}
                resizingColumn={''}
                columnRefs={new Map()}
                onDrop={jest.fn()}
            />,
        )
        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })

    test('should match snapshot, isSelected', async () => {
        // Sync
        FetchMock.fn.mockReturnValueOnce(FetchMock.jsonResponse(JSON.stringify([board, view, view2, card, cardTemplate])))
        FetchMock.fn.mockReturnValue(FetchMock.jsonResponse(JSON.stringify([{username: 'username_1'}, {username: 'username_2'}])))

        const boardTree = await MutableBoardTree.sync(board.id, view.id)
        expect(boardTree).toBeDefined()
        expect(FetchMock.fn).toBeCalledTimes(2)

        const cardTree = new MutableCardTree(card)

        const component = wrapProviders(
            <TableRow
                boardTree={boardTree!}
                cardTree={cardTree}
                card={card}
                isSelected={true}
                focusOnMount={false}
                onSaveWithEnter={jest.fn()}
                showCard={jest.fn()}

                readonly={false}
                offset={0}
                resizingColumn={''}
                columnRefs={new Map()}
                onDrop={jest.fn()}
            />,
        )
        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })

    test('should match snapshot, collapsed tree', async () => {
        // Sync
        view.collapsedOptionIds = ['value1']
        view.hiddenOptionIds = []

        FetchMock.fn.mockReturnValueOnce(FetchMock.jsonResponse(JSON.stringify([board, view, view2, card, cardTemplate])))
        FetchMock.fn.mockReturnValue(FetchMock.jsonResponse(JSON.stringify([{username: 'username_1'}, {username: 'username_2'}])))

        const boardTree = await MutableBoardTree.sync(board.id, view.id)
        expect(boardTree).toBeDefined()
        expect(FetchMock.fn).toBeCalledTimes(2)
        const cardTree = new MutableCardTree(card)

        const component = wrapProviders(
            <TableRow
                boardTree={boardTree!}
                cardTree={cardTree}
                card={card}
                isSelected={false}
                focusOnMount={false}
                onSaveWithEnter={jest.fn()}
                showCard={jest.fn()}

                readonly={false}
                offset={0}
                resizingColumn={''}
                columnRefs={new Map()}
                onDrop={jest.fn()}
            />,
        )
        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })

    test('should match snapshot, display properties', async () => {
        // Sync
        view.visiblePropertyIds = ['property1', 'property2']

        FetchMock.fn.mockReturnValueOnce(FetchMock.jsonResponse(JSON.stringify([board, view, view2, card, cardTemplate])))
        FetchMock.fn.mockReturnValue(FetchMock.jsonResponse(JSON.stringify([{username: 'username_1'}, {username: 'username_2'}])))

        const boardTree = await MutableBoardTree.sync(board.id, view.id)
        expect(boardTree).toBeDefined()
        expect(FetchMock.fn).toBeCalledTimes(2)
        const cardTree = new MutableCardTree(card)
        const component = wrapProviders(
            <TableRow
                boardTree={boardTree!}
                cardTree={cardTree}
                card={card}
                isSelected={false}
                focusOnMount={false}
                onSaveWithEnter={jest.fn()}
                showCard={jest.fn()}
                readonly={false}
                offset={0}
                resizingColumn={''}
                columnRefs={new Map()}
                onDrop={jest.fn()}
            />,
        )
        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })

    test('should match snapshot, resizing column', async () => {
        // Sync
        view.visiblePropertyIds = ['property1', 'property2']

        FetchMock.fn.mockReturnValueOnce(FetchMock.jsonResponse(JSON.stringify([board, view, view2, card, cardTemplate])))
        FetchMock.fn.mockReturnValue(FetchMock.jsonResponse(JSON.stringify([{username: 'username_1'}, {username: 'username_2'}])))

        const boardTree = await MutableBoardTree.sync(board.id, view.id)
        expect(boardTree).toBeDefined()
        expect(FetchMock.fn).toBeCalledTimes(2)
        const cardTree = new MutableCardTree(card)
        const component = wrapProviders(
            <TableRow
                boardTree={boardTree!}
                cardTree={cardTree}
                card={card}
                isSelected={false}
                focusOnMount={false}
                onSaveWithEnter={jest.fn()}
                showCard={jest.fn()}
                readonly={false}
                offset={0}
                resizingColumn={'property1'}
                columnRefs={new Map()}
                onDrop={jest.fn()}
            />,
        )
        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })
})
