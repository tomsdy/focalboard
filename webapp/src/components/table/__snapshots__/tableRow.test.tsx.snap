// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/table/TableRow should match snapshot 1`] = `
<div>
  <div
    class="TableRow octo-table-row"
    draggable="true"
    style="opacity: 1;"
  >
    <div
      class="octo-table-cell title-cell"
      id="mainBoardHeader"
      style="width: 100px;"
    >
      <div
        class="octo-icontitle"
      >
        <div
          class="octo-icon"
        >
          i
        </div>
        <input
          class="Editable undefined"
          placeholder="Untitled"
          spellcheck="true"
          title="title"
          value="title"
        />
      </div>
      <div
        class="open-button"
      >
        <button
          class="Button  "
          type="button"
        >
          Open
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/table/TableRow should match snapshot, collapsed tree 1`] = `
<div>
  <div
    class="TableRow octo-table-row hidden"
    draggable="true"
    style="opacity: 1;"
  >
    <div
      class="octo-table-cell title-cell"
      id="mainBoardHeader"
      style="width: 100px;"
    >
      <div
        class="octo-icontitle"
      >
        <div
          class="octo-icon"
        >
          i
        </div>
        <input
          class="Editable undefined"
          placeholder="Untitled"
          spellcheck="true"
          title="title"
          value="title"
        />
      </div>
      <div
        class="open-button"
      >
        <button
          class="Button  "
          type="button"
        >
          Open
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/table/TableRow should match snapshot, display properties 1`] = `
<div>
  <div
    class="TableRow octo-table-row hidden"
    draggable="true"
    style="opacity: 1;"
  >
    <div
      class="octo-table-cell title-cell"
      id="mainBoardHeader"
      style="width: 100px;"
    >
      <div
        class="octo-icontitle"
      >
        <div
          class="octo-icon"
        >
          i
        </div>
        <input
          class="Editable undefined"
          placeholder="Untitled"
          spellcheck="true"
          title="title"
          value="title"
        />
      </div>
      <div
        class="open-button"
      >
        <button
          class="Button  "
          type="button"
        >
          Open
        </button>
      </div>
    </div>
    <div
      class="octo-table-cell"
      style="width: 100px;"
    >
      <div
        class="ValueSelector css-2b097c-container"
      >
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-18140j1-Control"
        >
          <div
            class=" css-5conz1-ValueContainer"
          >
            <div
              class=" css-14tjw7f-singleValue"
            >
              <span
                class="Label color1 Label-single-select"
              >
                <span
                  class="Label-text"
                >
                  value 1
                </span>
              </span>
            </div>
            <div
              class="css-1shkodo-Input"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-2-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              class=" css-tpaeio-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                />
              </svg>
            </div>
            <span
              class=" css-43ykx9-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-19sxey8-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="octo-table-cell"
      style="width: 100px;"
    >
      <div
        class="ValueSelector css-2b097c-container"
      >
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        >
          <span
            id="aria-selection"
          />
          <span
            id="aria-context"
          >
              Select is focused ,type to refine list, press Down to open the menu, 
          </span>
        </span>
        <div
          class=" css-18140j1-Control"
        >
          <div
            class=" css-5conz1-ValueContainer"
          >
            <div
              class=" css-1wa3eu0-placeholder"
            />
            <div
              class="css-1shkodo-Input"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-3-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class=" css-43ykx9-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-hl9mox-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/table/TableRow should match snapshot, isSelected 1`] = `
<div>
  <div
    class="TableRow octo-table-row selected"
    draggable="true"
    style="opacity: 1;"
  >
    <div
      class="octo-table-cell title-cell"
      id="mainBoardHeader"
      style="width: 100px;"
    >
      <div
        class="octo-icontitle"
      >
        <div
          class="octo-icon"
        >
          i
        </div>
        <input
          class="Editable undefined"
          placeholder="Untitled"
          spellcheck="true"
          title="title"
          value="title"
        />
      </div>
      <div
        class="open-button"
      >
        <button
          class="Button  "
          type="button"
        >
          Open
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/table/TableRow should match snapshot, read-only 1`] = `
<div>
  <div
    class="TableRow octo-table-row"
    draggable="true"
    style="opacity: 1;"
  >
    <div
      class="octo-table-cell title-cell"
      id="mainBoardHeader"
      style="width: 100px;"
    >
      <div
        class="octo-icontitle"
      >
        <div
          class="octo-icon"
        >
          i
        </div>
        <input
          class="Editable readonly undefined"
          placeholder="Untitled"
          readonly=""
          spellcheck="true"
          title="title"
          value="title"
        />
      </div>
      <div
        class="open-button"
      >
        <button
          class="Button  "
          type="button"
        >
          Open
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/table/TableRow should match snapshot, resizing column 1`] = `
<div>
  <div
    class="TableRow octo-table-row hidden"
    draggable="true"
    style="opacity: 1;"
  >
    <div
      class="octo-table-cell title-cell"
      id="mainBoardHeader"
      style="width: 100px;"
    >
      <div
        class="octo-icontitle"
      >
        <div
          class="octo-icon"
        >
          i
        </div>
        <input
          class="Editable undefined"
          placeholder="Untitled"
          spellcheck="true"
          title="title"
          value="title"
        />
      </div>
      <div
        class="open-button"
      >
        <button
          class="Button  "
          type="button"
        >
          Open
        </button>
      </div>
    </div>
    <div
      class="octo-table-cell"
      style="width: 100px;"
    >
      <div
        class="ValueSelector css-2b097c-container"
      >
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-18140j1-Control"
        >
          <div
            class=" css-5conz1-ValueContainer"
          >
            <div
              class=" css-14tjw7f-singleValue"
            >
              <span
                class="Label color1 Label-single-select"
              >
                <span
                  class="Label-text"
                >
                  value 1
                </span>
              </span>
            </div>
            <div
              class="css-1shkodo-Input"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-4-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <div
              aria-hidden="true"
              class=" css-tpaeio-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M14.348 14.849c-0.469 0.469-1.229 0.469-1.697 0l-2.651-3.030-2.651 3.029c-0.469 0.469-1.229 0.469-1.697 0-0.469-0.469-0.469-1.229 0-1.697l2.758-3.15-2.759-3.152c-0.469-0.469-0.469-1.228 0-1.697s1.228-0.469 1.697 0l2.652 3.031 2.651-3.031c0.469-0.469 1.228-0.469 1.697 0s0.469 1.229 0 1.697l-2.758 3.152 2.758 3.15c0.469 0.469 0.469 1.229 0 1.698z"
                />
              </svg>
            </div>
            <span
              class=" css-43ykx9-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-19sxey8-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="octo-table-cell"
      style="width: 100px;"
    >
      <div
        class="ValueSelector css-2b097c-container"
      >
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        >
          <span
            id="aria-selection"
          />
          <span
            id="aria-context"
          >
              Select is focused ,type to refine list, press Down to open the menu, 
          </span>
        </span>
        <div
          class=" css-18140j1-Control"
        >
          <div
            class=" css-5conz1-ValueContainer"
          >
            <div
              class=" css-1wa3eu0-placeholder"
            />
            <div
              class="css-1shkodo-Input"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-5-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class=" css-43ykx9-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-hl9mox-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
